### PR TITLE
Search page cleanup

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
For #640 

### Description (What does it do?)
This PR makes several changes to the search page:

1. Removes the "Active Facets" display above facet list. ([These chips](https://github.com/mitodl/mit-open/assets/9010790/d38b271a-20e1-43f7-bf6d-8383cc05648b) have been removed.)
2. Now, if a facet is active, its checkbox should always be shown.
     - Facets options are only included in the API results if their result count is nonzero. So if an active facet corresponds to zero results, it previously was not shown. This situation can occur by toggling a facet on, then entering query text. For example: https://mitopen-rc.odl.mit.edu/search/?q=quaternions&t=Visual+Arts
3. The results title count has been removed. (No more '524 results for "Physics".)
4. "Clear all facets" no longer clears the resource tab selection.
5. Adds "Offered by" facet.
6. Facets now use API query names ("department" not "d")
7. Facets now work even if they aren't explicitly included in UI.
    - e.g., `http://localhost:8063/search?platform=mitx` 

### Screenshots (if appropriate):
Screenshot 2024-03-25 at 3 01 27 PM

Screenshot 2024-03-25 at 2 50 42 PM

### How can this be tested?
1. Visit http://localhost:8063/search
2. Try:
    - click resource tabs
    - entering query text
    - toggling left sidebar filters on and off
    - using the "Clear all" filters button

    Things to look out for
    - URL bar should use "real" API parameter names ("department" not "d")
3. Append a filter not included in the UI to the URL; for example `?platform=mitxonline` or (`&platform=mitxonline`). The results should update.

### CODE REVIEW

File: frontends/mit-open/src/pages/SearchPage/SearchPage.tsx [hunk 1]
```diff
@@ -222,47 +232,95 @@ const TABS: TabConfig[] = [
...
 
+/**
+ * Augment the facet buckets for `groupKey` with active values that have no
+ * results.
+ */
+const includeActiveZerosInBuckets = (
+  groupKey: string,
+  aggregations: LearningResourceSearchResponse["metadata"]["aggregations"],
+  params: LRSearchRequest,
+) => {
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/pages/SearchPage/SearchPage.tsx:243]: @abeglova Maybe the `FacetDisplay` component should handle this? OCW does seem to add-in active facets with zero count, but I don't think it's in course-search-utils currently?

**Review by abeglova — CHANGES_REQUESTED** [review]
This works great!

I agree with you that includeActiveZerosInBuckets should be moved to course-search-utils. 

Additionally it looks like you implemented useValidatedSearchParams in course-search-utils but are not using it in this pr - I can search for an invalid param such as http://localhost:8063/search/?topic=monkey and it adds monkey to the topics list

**Comment by ChristopherChudzicki** [comment]
> Additionally it looks like you implemented useValidatedSearchParams in course-search-utils but are not using it in this pr - I can search for an invalid param such as http://localhost:8063/search/?topic=monkey and it adds monkey to the topics list

`useValidatedSearchParams` isn't actually exported from course-search-utils. But it is used in implementing `useResourceSearchParams` and `useContentFileSearchParams`. For example:

```ts
const useResourceSearchParams = ({
  searchParams,
  setSearchParams,
  facets = [],
  onFacetsChange
}: UseValidatedSearchParamsProps<ResourceSearchRequest>) => {
  return useValidatedSearchParams<ResourceSearchRequest>({
    searchParams,
    setSearchParams,
    facets,
    validators: resourceSearchValidators,
    onFacetsChange
  })
}
```

> I can search for an invalid param such as http://localhost:8063/search/?topic=monkey and it adds monkey to the topics list

The validation is based on the OpenAPI spec, by which `"monkey"` is a valid topic. (For topic, the spec just says "string"; department and platform, we have an enum; similarly https://mitopen-rc.odl.mit.edu/api/v1/learning_resources_search/?topic=monkey is a 200 response with zero hits, whereas https://mitopen-rc.odl.mit.edu/api/v1/learning_resources_search/?department=monkey is a 400.)  My preference would be to continue validating based only on the OpenAPI spec...It's not detrimental to the UI (since filter choices are populated off the API response) and keeps things simple.

File: frontends/mit-open/src/pages/SearchPage/SearchPage.tsx (hunk 1 above)
Comment (ChristopherChudzicki) [reply frontends/mit-open/src/pages/SearchPage/SearchPage.tsx:243]: Sounds good. I'd like to do that separately from https://github.com/mitodl/course-search-utils/pull/78, but I can do it before merging this PR.

https://github.com/mitodl/course-search-utils/pull/78 is big enough already, and I'd like replace the Enzyme-based facet tests if I'm going to be adding more there.

**Review by abeglova — APPROVED** [review]

**Comment by Ferdi** [comment]
@ChristopherChudzicki if it's not too late can we change the order of tabs to all, courses, programs, podcasts ?

**Comment by ChristopherChudzicki** [comment]
> @ChristopherChudzicki if it's not too late can we change the order of tabs to all, courses, programs, podcasts ?
Screenshot 2024-03-27 at 11 25 45 AM

**Review by abeglova — APPROVED** [review]

**Pull request merged**